mixer/v1/config/client/auth.proto

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017 Istio Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1.config.client;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+message AuthPolicy {
+  // APIKey defines configuration for generating the API key attribute
+  // from HTTP requests.
+  //
+  // See https://swagger.io/docs/specification/authentication/api-keys
+  // for a general overview of API keys as defined by OpenAPI.
+  message APIKey {
+    oneof key {
+      // API Key is sent as a query parameter. `query` represents the
+      // query string parameter name.
+      //
+      // For example, `query=api_key` should be used with the
+      // following request:
+      //
+      //     GET /something?api_key=abcdef12345
+      //
+      string query = 1;
+      // API key is send in a request header. `header` represents the
+      // header name.
+      //
+      // For example, `header=X-API-KEY` should be used with the
+      // following request:
+      //
+      //     GET /something HTTP/1.1
+      //     X-API-Key: abcdef12345
+      //
+      string header = 2;
+
+      // API key is sent in a
+      // [cookie](https://swagger.io/docs/specification/authentication/cookie-authentication),
+      //
+      // For example, `cookie=X-API-KEY` should be used for the
+      // following request:
+      //
+      //     GET /something HTTP/1.1
+      //     Cookie: X-API-KEY=abcdef12345
+      //
+      string cookie = 3;
+    }
+
+    // Attribute name to use when forwarding api_key to mixer,
+    // e.g. api.key
+    string attribute_name = 4;
+  }
+
+  message JWT {
```

Comment:
I think JWT has two parts:
1) JWT definition which is global per service,  to define some config for each issuer.   such as uri to fetch the public key.
2) JWT requirement which is subject to attribute match.    it will just points to a list of issuers and its audiences

Reply:
google/api/auth.proto's [AuthProvider](https://github.com/googleapis/googleapis/blob/master/google/api/auth.proto#L85) defines issuer and audiences alongside jwks_uri. I reworked the JWT definition to match. Let me know if this makes sense.